Content.Server/_RMC14/Rules/CMDistressSignalRuleSystem.cs

```diff
@@ -419,7 +422,28 @@ private void OnPlayerSpawning(PlayerSpawningEvent ev)
         var query = QueryActiveRules();
         while (query.MoveNext(out _, out _, out var comp, out _))
         {
-            if (GetSpawner(comp, job) is not { } spawnerInfo)
+            if (comp.MaxJobsPerSquad.Count == 0 && comp is CMDistressSignalRuleComponent)
+            {
+                var allJobs = _stationJob.GetRoundStartJobs(ev.Station!.Value);
```

Comment:
This will also error if station is null, nullable warnings need to be handled.
This will also not work for dynamic job slots based on player count later, GetJobs should be used instead but you would need to do this earlier, in the handler for rulePlayerSpawningEvent for example.